src/main/java/bitflow4j/steps/fork/distribute/ForkTemplate.java

```diff
@@ -0,0 +1,139 @@
+package bitflow4j.steps.fork.distribute;
+
+import bitflow4j.Pipeline;
+import bitflow4j.Sample;
+import bitflow4j.misc.Pair;
+import bitflow4j.script.registry.Description;
+import bitflow4j.steps.fork.ScriptableDistributor;
+
+import java.io.IOException;
+import java.util.*;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+import java.util.stream.Collectors;
+
+@Description("A Fork that forks the samples based on the provided template. Samples with the same template values of the provided" +
+        " template name will be passed to the same fork. Use ${tagName} Syntax to evaluate the given tags value.")
+public class ForkTemplate implements ScriptableDistributor {
```

Comment:
It looks like almost this entire class is a duplicate of ForkTag.
I suggest making ForkTag a subclass of this, and convert the single tag that it receives in the constructor to the according template (surround it with '${}'). This keeps all functionality, while removing all duplications.